sql/moz-fx-data-shared-prod/stripe/itemized_payout_reconciliation/view.sql

```diff
@@ -0,0 +1,87 @@
+CREATE OR REPLACE VIEW
+  `moz-fx-data-shared-prod.stripe.itemized_payout_reconciliation`
+AS
+WITH customers AS (
+  SELECT
+    id,
+    address.country,
+    NULLIF(address.postal_code, "") AS postal_code,
+    NULLIF(address.state, "") AS state,
+  FROM
+    `moz-fx-data-shared-prod`.stripe_external.customers_v1
+  WHERE
+    address.country IN ("US", "CA")
+),
+postal_code_to_state AS (
+  SELECT
+    country,
+    postal_code,
+    IF(
+      COUNT(DISTINCT state) > 1,
+      ERROR("postal code resolved to mulitple states: " || TO_JSON_STRING(postal_code)),
+      ANY_VALUE(state)
+    ) AS state,
+  FROM
+    customers
+  WHERE
+    postal_code IS NOT NULL
+  GROUP BY
+    country,
+    postal_code
+),
+charges AS (
+  SELECT
+    id,
+    payment_method_details.card.country,
+    NULLIF(billing_details.address.postal_code, "") AS postal_code,
+    state,
+  FROM
+    `moz-fx-data-shared-prod`.stripe_external.charges_v1
+  JOIN
+    postal_code_to_state
+  ON
+    payment_method_details.card.country = postal_code_to_state.country
+    AND billing_details.address.postal_code = postal_code_to_state.postal_code
+  WHERE
+    payment_method_details.card.country IN ("US", "CA")
+),
+enriched AS (
+  SELECT
+    report.*,
+    charges.state,
+    charges.postal_code,
```

Comment:
see https://mozilla-hub.atlassian.net/browse/DVPN-111?focusedCommentId=517297 for discussion on how these fields are being calculated